### PR TITLE
Change log level of methods name logged in Scaffolding

### LINKED ISSUE #1702: Use trace level logging in generated code
Is it possible to not have one function log every time its called?

I have a very simple function 

```rust
#[uniffi::export]
pub fn pod_restart_count(pod: Pod) -> i32 {
    pod.total_restart_count()
}
```

Its called a lot of times and drowns out all the other logs, would it be possible to opt out of logging on a per function basis?
Screenshot 2023-08-10 at 6 45 24 PM

### PR DESCRIPTION
**Pull request by Vinnstah**
Fixes #1702 

Change the log level to trace for method names logged by UniFFI internals when FFI calls into Rust. 

Augmented existing logging tests in Python, asserting that the changed level works as intended.

Let me know if you want me to put the tests in a **new** fixture instead of recycling an existing one.

### CODE REVIEW

File: fixtures/regressions/logging-callback-interface/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "uniffi-fixture-logging-callback-interface"
 edition = "2021"
+authors = ["Firefox Sync Team <sync-team@mozilla.com>", "Viktor Jansson <v.l.jansson@gmail.com>"]
```
Comment (Vinnstah) [inline fixtures/regressions/logging-callback-interface/Cargo.toml:4]: Please let me know if you want me to revert this change. I git blamed and saw that Ben has contributed previously, let me know if you want me to add him as well (I think Mark is a member of the sync team?).

File: fixtures/regressions/logging-callback-interface/src/lib.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@ use std::sync::{Once, RwLock};
 
 // Logger trait that the foreign code implements
 pub trait Logger: Sync + Send {
-    fn log_message(&self, message: String);
+    fn log_message(&self, message: String, level: String);
```
Comment (Vinnstah) [inline fixtures/regressions/logging-callback-interface/src/lib.rs:9]: String is more clear than `u8`, and seems like overkill introducing a UniFFI exported Enum.

File: uniffi_macros/src/export/scaffolding.rs
Comment (Vinnstah) [inline uniffi_macros/src/export/scaffolding.rs:1]: The actual fix of #1702

File: fixtures/regressions/logging-callback-interface/src/test.udl
Comment (Vinnstah) [inline fixtures/regressions/logging-callback-interface/src/test.udl:1]: Will make sense when you read `test.py`...

**Review by Vinnstah — COMMENTED** [review]
Left some clarifying comments.

**Review by Sajjon — APPROVED** [review]
Nice, I've been waiting for this fix actually, it will be most useful for me in [Sargon](https://github.com/radixdlt/sargon/pull/131), where I've recently worked with logging.

File: fixtures/regressions/logging-callback-interface/Cargo.toml (hunk 1 above)
Comment (mhammond) [reply fixtures/regressions/logging-callback-interface/Cargo.toml:4]: Ben's also in the sync team - we'd also be up for removing this line entirely, but until then I think this is fine.

File: uniffi_macros/src/export/scaffolding.rs [hunk 3]
```diff
@@ -285,7 +285,7 @@ pub(super) fn gen_ffi_function(
             #[doc(hidden)]
             #[no_mangle]
             pub extern "C" fn #ffi_ident(#(#param_names: #param_types,)*) -> ::uniffi::Handle {
-                ::uniffi::deps::log::debug!(#name);
+                ::uniffi::deps::log::trace!(#name);
```
Comment (mhammond) [inline uniffi_macros/src/export/scaffolding.rs:288]: I guess I don't really understand why most of the above changes - ie, I'm not sure how exactly the fixture changes really ensures these 2 lines actually work. Or to put it yet another way, I think I'd r+ this with just these 2 lines.

Comment (mhammond) [reply uniffi_macros/src/export/scaffolding.rs:288]: I'm not quite sure what you are saying there, but my point is that unless I'm still missing something, I think the test code here is testing the log crate rather than the uniffi crate?

Comment (Sajjon) [reply uniffi_macros/src/export/scaffolding.rs:288]: The changed code in the test **piggybacks** on an existing test, but it does **indeed** test uniffi crate.

It tests the changed UniFFI source code. Without the changed source code the test would fail. 

The test ensures the level at which UniFFI source code logs name of methods is trace.

Comment (mhammond) [reply uniffi_macros/src/export/scaffolding.rs:288]: > Without the changed source code the test would fail.

I hope it's clear that I do infact understand that.

Comment (Sajjon) [reply uniffi_macros/src/export/scaffolding.rs:288]: @Vinnstah I would benefit from this PR getting landed :D 

Sounds like Mark would prefer if the changes in the test be reverted, maybe you can revert and this might get merged :)

Comment (mhammond) [reply uniffi_macros/src/export/scaffolding.rs:288]: To be clear, I'm only weakly against tests for this - IMO the value is lower than the cost. But I'm open to be convinced otherwise, and it's difficult to argue too strongly *against* new tests really :)

**Comment by mhammond** [comment]
Hi, is there any interest in removing some of the test code here? If I don't hear back soon, I'll put up the 2-line PR that actually fixes this and close this, but I'd like to offer the opportunity to have it completed here.

**Comment by Vinnstah** [comment]
Hi Mark,
Apologies for the delay. I'll fix the tests and push a new commit asap.

**Comment by praveenperera** [comment]
thank you for this! @Vinnstah !

**Comment by bendk** [comment]
I didn't realize this issue has been sitting in the queue for so long.  Does https://github.com/mozilla/uniffi-rs/pull/2211 handle your needs or is there still work to be done here?

**Comment by bendk** [comment]
I should have pointed you to https://github.com/mozilla/uniffi-rs/pull/2275.  As far as I know, there are no more log calls in the current code.  I don't think we even have a dependency on the crate.  Those calls were replaced by an internal logging mechanism, that's disabled by default.

**Comment by praveenperera** [comment]
@bendk I'm using master now and its working, no more noisy logs thank you!

**Comment by mhammond** [comment]
We've recently removed all `log` calls from the generated scaffolding.
